software/firmware/configuration.py

```diff
@@ -44,6 +44,71 @@ def validate(self, value) -> Validation:
         raise NotImplementedError
 
 
+class FloatConfigPoint(ConfigPoint):
+    """A `ConfigPoint` that requires the selection from a range of floats.  The default value
+    must lie within the specified range
+
+    :param name: The name of this `ConfigPoint`, will be used by scripts to lookup the configured value.
+    :param minimum: The minimum allowed value
+    :param maximum: The maximum allowed value
+    :param default: The default value
+    """
+
+    def __init__(self, name: str, minimum: float, maximum: float, default: float):
+        super().__init__(name=name, type=float, default=default)
+        self.maximum = maximum
+        self.minimum = minimum
+
+        if default < minimum or default > maximum:
+            raise Exception(f"Defaulf {default} is out of range")
+
+    def validate(self, value) -> Validation:
+        """Validates the given value with this ConfigPoint. Returns a `Validation` containing the
+        validation result, as well as an error message containing the reason for a validation failure.
+        """
+        # Silently caste integers into floats, so we don't crash if someone writes e.g. "0" instead of "0.0"
+        if type(value) is int:
+            value = float(value)
```

Comment:
Because all of the other `ConfigPoint` classes have an explicit type in the name, keeping the floating-point one called `FloatConfigPoint` improves clarity. And I can definitely see cases where being able to specify between floats or ints would be desirable.

That extra if statement is really only there to handle the JSON parsing, where users may enter e.g. `0` instead of `0.0`, and Python will load the JSON field as an integer.  It's mostly there as a convenience to keep the file processing a little easier on the user.